src/main/java/org/apache/xml/security/utils/KeyUtils.java

```diff
@@ -0,0 +1,280 @@
+                LOG.log(Level.WARNING, "EC keys are detected for key agreement algorithm! " +
+                        "Cryptographic algorithm may not be secure, consider using a different algorithm (and keys). ");
+            }
+            algorithm = algorithm + (algorithm.equalsIgnoreCase("EC") ? "DH" : "");
+            KeyAgreement keyAgreement = KeyAgreement.getInstance(algorithm);
+            keyAgreement.init(privateKey);
+            keyAgreement.doPhase(publicKey, true);
+            byte[] secret = keyAgreement.generateSecret();
+            byte[] kek = deriveKeyEncryptionKey(secret, parameterSpec.getKeyDerivationParameter());
+            return new SecretKeySpec(kek, "AES");
+
+
+        } catch (XMLSecurityException | NoSuchAlgorithmException | InvalidKeyException e) {
+            throw new XMLEncryptionException(e);
+        }
+    }
+
+    /**
+     * Defines the key size for the encrypting algorithm.
+     *
+     * @param keyWrapAlg the key wrap algorithm URI
+     * @return the key size in bits
+     * @throws XMLEncryptionException if the key wrap algorithm is not supported
+     */
+    public static int getAESKeyBitSizeForWrapAlgorithm(String keyWrapAlg) throws XMLEncryptionException {
+        switch (keyWrapAlg) {
+            case EncryptionConstants.ALGO_ID_KEYWRAP_AES128:
+                return 128;
+            case EncryptionConstants.ALGO_ID_KEYWRAP_AES192:
+                return 192;
+            case EncryptionConstants.ALGO_ID_KEYWRAP_AES256:
+                return 256;
+            default:
+                throw new XMLEncryptionException("Unsupported KeyWrap Algorithm");
+        }
+    }
+
+
+    /**
+     * Derive a key encryption key from a shared secret and keyDerivationParameter. Currently only the ConcatKDF is supported.
+     * @param sharedSecret the shared secret
+     * @param keyDerivationParameter the key derivation parameters
+     * @return the derived key encryption key
+     * @throws XMLSecurityException if the key derivation algorithm is not supported
+     */
+    public static byte[] deriveKeyEncryptionKey(byte[] sharedSecret, KeyDerivationParameter keyDerivationParameter)
+            throws XMLSecurityException {
+        int iKeySize = keyDerivationParameter.getKeyBitLength()/8;
+        String keyDerivationAlgorithm = keyDerivationParameter.getAlgorithm();
+        if (!EncryptionConstants.ALGO_ID_KEYDERIVATION_CONCATKDF.equals(keyDerivationAlgorithm)) {
+            throw new XMLEncryptionException( "unknownAlgorithm",
+                    keyDerivationAlgorithm);
+        }
+        ConcatKeyDerivationParameter ckdfParameter = (ConcatKeyDerivationParameter) keyDerivationParameter;
+
+
+        // get parameters
+        String digestAlgorithm =ckdfParameter.getDigestAlgorithm();
+        if (digestAlgorithm == null) {
+            LOG.log(Level.DEBUG, "Missing ConcatKDFParams Digest algorithm for key derivation with algorithm [{0}]", keyDerivationAlgorithm);
```

Comment:
see above

Reply:
Removed the constructor with default digest